### PR TITLE
feat: [PIE-3671]: Test case for common component RepositorySelect

### PR DESCRIPTION
**Pull request by depanshu-dhiman**
##### Summary:
Test cases for common component RepositorySelect

##### Jira Links:
https://harness.atlassian.net/browse/PIE-3671

##### Screenshots:
Screenshot 2022-06-20 at 4 17 43 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/10-common/components/RepositorySelect/__tests__/RepositorySelect.test.tsx [hunk 1]
```diff
@@ -36,15 +59,153 @@ describe('RepositorySelect test', () => {
...
+      error: {
+        data: {
+          responseMessages: ['error']
+        }
+      },
+      loading: false
+    }))
+
+    const { container, getByText } = render(
+      <TestWrapper path={testPath} pathParams={pathParams}>
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/RepositorySelect/__tests__/RepositorySelect.test.tsx:118]: Same as above.

File: src/modules/10-common/components/RepositorySelect/__tests__/RepositorySelect.test.tsx [hunk 2]
```diff
@@ -36,15 +59,153 @@ describe('RepositorySelect test', () => {
...
+    expect(fetchRepos).toBeCalledTimes(2)
+  })
+
+  test('Changing repo will call onchange handler with the selectedValue & selectOptions and reset error', async () => {
+    ;(useGetListOfReposByRefConnector as jest.Mock).mockImplementation(() => {
+      return { data: mockReposSingle, refetch: fetchRepos }
+    })
+
+    render(
+      <TestWrapper path={testPath} pathParams={pathParams}>
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/RepositorySelect/__tests__/RepositorySelect.test.tsx:144]: Same as above.

File: src/modules/10-common/components/RepositorySelect/__tests__/RepositorySelect.test.tsx [hunk 3]
```diff
@@ -36,15 +59,153 @@ describe('RepositorySelect test', () => {
...
+        </Formik>
+      </TestWrapper>
+    )
+
+    // test defaultSelected if only 1 item in list
+    expect(setErrorResponse).not.toBeCalled()
+    expect(repoChangehandler).toHaveBeenLastCalledWith({ label: 'repo1', value: 'repo1' }, [])
+
+    // multiple items now
+    ;(useGetListOfReposByRefConnector as jest.Mock).mockImplementation(() => {
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/RepositorySelect/__tests__/RepositorySelect.test.tsx:161]: Write this in separate test case

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/10-common/components/RepositorySelect/__tests__/RepositorySelect.test.tsx [hunk 4]
```diff
@@ -36,15 +59,153 @@ describe('RepositorySelect test', () => {
...
       </TestWrapper>
     )
     await waitFor(() => expect(fetchRepos).toBeCalledTimes(1))
     expect(getByText('repository')).toBeInTheDocument()
     expect(container).toMatchSnapshot()
   })
+
+  test('Repository list api called only if connectorRef is provided and field is not disabled', () => {
+    render(
+      <TestWrapper path={testPath} pathParams={pathParams}>
```
[the remark this answers is not in the document]
Comment (depanshu-dhiman) [reply src/modules/10-common/components/RepositorySelect/__tests__/RepositorySelect.test.tsx:74]: I have tried to remove it. But the component relies on `useParams` and we can either mock `useParams` or use `TestWrapper`. 
I was able to remove `Formik`. But `TestWrapper` seems mandatory

File: src/modules/10-common/components/RepositorySelect/__tests__/RepositorySelect.test.tsx (hunk 3 above)
Comment (depanshu-dhiman) [reply src/modules/10-common/components/RepositorySelect/__tests__/RepositorySelect.test.tsx:152]: done

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Pull request merged**